bioio_base/reader.py

```diff
@@ -1070,6 +1083,89 @@ def get_xarray_dask_stack(self, **kwargs: Any) -> xr.DataArray:
         """
         return transforms.generate_stack(self, mode="xarray_dask_data", **kwargs)
 
+    # --- New embedded metadata properties for the manifest ---
+    def _get_dim_size(self, dim: str) -> Optional[int]:
+        if dim in self.dims.order:
+            idx = self.dims.order.index(dim)
+            return self.shape[idx]
+        return None
+
+    @property
+    @embedded_metadata(MetadataLabels.DIMENSIONS_PRESENT.value)
+    def dimensions_present(self) -> Optional[str]:
+        return self.dims.order
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_C.value)
+    def image_size_c(self) -> Optional[int]:
+        return self._get_dim_size("C")
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_T.value)
+    def image_size_t(self) -> Optional[int]:
+        return self._get_dim_size("T")
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_X.value)
+    def image_size_x(self) -> Optional[int]:
+        return self._get_dim_size("X")
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_Y.value)
+    def image_size_y(self) -> Optional[int]:
+        return self._get_dim_size("Y")
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_Z.value)
+    def image_size_z(self) -> Optional[int]:
+        return self._get_dim_size("Z")
+
+    @property
+    @embedded_metadata(MetadataLabels.PIXEL_SIZE_X.value)
+    def pixel_size_x(self) -> Optional[float]:
+        return self.physical_pixel_sizes.X
+
+    @property
+    @embedded_metadata(MetadataLabels.PIXEL_SIZE_Y.value)
+    def pixel_size_y(self) -> Optional[float]:
+        return self.physical_pixel_sizes.Y
+
+    @property
+    @embedded_metadata(MetadataLabels.PIXEL_SIZE_Z.value)
+    def pixel_size_z(self) -> Optional[float]:
+        return self.physical_pixel_sizes.Z
+
+    @property
+    @embedded_metadata(MetadataLabels.TIMELAPSE.value)
+    def timelapse(self) -> Optional[bool]:
+        return self.image_size_t > 0 if self.image_size_t is not None else None
+
+    # -----------------------------------------------------------------------------
+    # Manifest property that dynamically builds a dictionary of embedded metadata.
+    # -----------------------------------------------------------------------------
+    @property
+    def manifest(self) -> Dict[str, Any]:
```

Comment:
I think we should type the return value better

Reply:
We could create a type like EmbeddedMetadata thats a single key and have it be something like List [EmbeddedMetadata]. Dont really have an opinion about the typing here.

Reply:
With the current implementation it seems impossible to type it more narrowly: subclasses can add arbitrary properties without overriding `manifest`